interview-questions.md

```diff
@@ -12,46 +12,73 @@ Your answer:
 
-Researched answer:
+Researched answer: 
+-Git is a version control system. This helps track changes made in code, offers the capability of branches, ensures faster/more efficient ways to manage coding in projects of alls sizes. Git also features the capability of having multiple branches that are not connected to one another, so modifying/deleting one branch will not affect another.(https://git-scm.com/)
+-GitHub is a developer platform that is open source, allows developers to build projects, edit them and share them. It also offers resources such as "how-to" guides, templates for different languages and extensions to download to assist users. A project can be made public or private. For a profile, you can receive access to more features and applications if you pay a monthly fee on a tiered pricing scale, however there is a free version. A user's profile will track commits, pushes ando other contributions made. A profile also has a "feed" that shows their activity in repositories and contributions, followers, and most actively used repositories.
 
 2. Which JavaScript operators will return a Boolean value?
 
-Your answer:
+Your answer: = , ===, /=, >=, <=. 
 
-Researched answer:
+Researched answer: Answer also includes || and &&. 
+    -Researched and summarized from "https://www.w3schools.com/js/js_booleans.asp"
+    - Boolean values dictate whether a variable or expression is true/false. They are utilized in comparison and conditionals statements that allow the output to show if the expression/variable entered is true or false. True or false can look like 1 === 1 (true) or 2 > 105(false). If an boolean variable has a value, it is true and if it does not have a value it is false (w3schools.com/js/js_booleans.asp). Boolean will also return false if it is undefined, Nan, Null, 0, or -0.(https://www.tutorialsteacher.com/javascript/javascript-boolean)
 
-1. What is the difference between an array's index and its value?
+3. What is the difference between an array's index and its value?
+
+Your answer: An array's value is stored in a single variable, so it will be an entire index.
+    Example. ["computer", "cat", "dinner", "trees"]
+        Index:   0          1        2         3
+    In a regular string, each character of the value would be it's own index.
+    Example: ("puppies)
+    Index: 6- 0/p, 1/u, 2/p, etc.
+   
+
+Researched answer: 
+-"The index of a value in an array, is that value's location within an array" (https://problemsolvingwithpython.com/05-NumPy-and-Arrays/05.05-Array-Indexing/#:~:text=The%20index%20of%20a%20value,in%20the%20code%20section%20below.&text=The%20array%20above%20contains%20three%20values%3A%202%20%2C%204%20and%206%20.) Each element in the array list can be accessed through indexing. Essentially, the index is the address of a value. Since array values are arranged like a list, the index will give you the location of a value on that list.
```

Comment:
Yes, the array's index is the address of a value! Each value in an array has its own index. The difference between the index and value, is that the index is the position of a value, whereas the value is the actual element that is at that position.